packages/components/src/modal/Modal.tsx

```diff
@@ -0,0 +1,73 @@
+import { Button } from "@/button";
+import { ButtonContainer, IconWrapper, darkTheme as buttonDarkTheme } from "@/button/Button.styled";
+import { type ButtonProps } from "@/button/Button.types";
+
+import { globalStyles } from "@/index.styled";
+import { darkTheme, ModalContainer, HeaderWrapper, BodyWrapper, ButtonsWrapper } from "@/modal/Modal.styled";
+import { type ModalProps } from "@/modal/Modal.types";
+import { useState } from "react";
+import LightModeCross from "../assets/Lightmodecross.svg"
+import DarkModeCross from "../assets/Darkmodecross.svg"
+
+
+const CrossButton = ({type, theme, icon, onClick}: ButtonProps) => {
+
+  // Apply global styles
+  globalStyles();
+
+  return (
+    <ButtonContainer 
+      className={theme === "dark" ? buttonDarkTheme : ''} 
+      onClick={onClick} 
+      type={type}
+    >
+      {icon && (
+        <IconWrapper>
+          {icon}
+        </IconWrapper>
+      )}
+    </ButtonContainer>
+  )
+}
+
+export const Modal = ({title, text, theme, option}: ModalProps) => {
+  
+  // Apply global styles
+  globalStyles();
+
+  const [open, setOpen] = useState(true);
+
+  const closeModal = () => {
+    setOpen(false);
+  }
+
+  return (
+    <ModalContainer 
```

Comment:
To improve accessibility, consider using the <dialog> tag.👍 And I see you haven't created the button to open the dialog yet, so please implement that as well 🥺 

Reference: https://ui.shadcn.com/docs/components/dialog